subprojects/studio-plugin/src/main/java/org/gradle/profiler/studio/plugin/system/GradleProfilerGradleSyncListener.java

```diff
@@ -48,6 +53,8 @@ public GradleSyncResult waitAndGetResult() {
             return results.take();
         } catch (InterruptedException e) {
             throw new RuntimeException(e);
+        } finally {
+            connection.disconnect();
```

Comment:
Seems like a code smell to close a connection in a place other than where it was opened. Instead in `AndroidStudioSystemHelper.doGradleSync()` I'd inline `subscribeToGradleSync()` and put a `try` around the `connection` object there.

Reply:
We can also do that, done